### DISCUSSION TITLE
HTML output numbered figures and tables without using crossref?

### DISCUSSION THREAD

**Discussion by davidbudzynski** [quarto-dev/quarto-cli#5775] [Q&A] [answered]
### Description

Is is possible to add numbers to figures and tables in html output without adding crossreff at all? I'm looking into using quarto as a replacement of `bookdown::html_document2` and knowing if I can achieve this feature would be very useful.

**Comment by cderv** [comment, marked as answer]
> Is is possible to add numbers to figures and tables in html output without adding crossreff at all?

What do you mean by not adding  crossref at all ? 

You can have numbers to figure and tables by adding the correct prefix in the label i.e `fig-`

````markdown
---
title: Figure
format: html
---

```{r}
#| label: fig-pressure
#| fig-cap: "Vapor Pressure of Mercury as a Function of Temperature"

plot(pressure)
```

````

This is indeed related to cross ref feature (https://quarto.org/docs/authoring/figures.html#cross-references) but no need to create reference to the figure in other place. It will just trigger the labelling and numbering.

**Reply by davidbudzynski** [reply]
My undertstanding was that numbering of tables and figures was only triggered when adding label and then mentioning the object in the text using crossref, like `lorem ipsum @my-figure-label.`, but I actually didn't try only adding the label to see what happens. I can confirm this works as expected. thanks! 

Is there an option in the yaml header that would elimate the necessity of adding labels to tables/plots altogether and add `Figure/Table X` to every table or figure in the document?

**Reply by mcanouil** [reply]
There is not such feature at the moment.
When using `knitr`, a hook that prepend `fig-` to the label could in theory be used.

**Reply by cderv** [reply]
Or using a Lua filter to add prefix to all images or tables. I think this would work.

**Reply by davidbudzynski** [reply]
I don't tink that this will be necessary in my case, I was just wondering if this is feature exists.
